api/namex/models/payment.py

```diff
@@ -62,6 +62,29 @@ def save_to_db(self):
     def delete_from_db(self):
         pass
 
-
-
-
+@event.listens_for(Payment, 'after_insert')
+@event.listens_for(Payment, 'before_update')
+def update_nr_state(mapper, connection, target):
+    """Set the state of the NR based on payment_status_code."""
+    from namex.models import Request
+
+    payment = target
+    nr = Request.find_by_id(payment.nrId)
+    if nr:
+        # could not make this update properly via the model so used raw sql
+        if payment.payment_status_code == 'COMPLETED':

```

Comment:
For the scenarios like upgrade to priority, reapply, extension where a payment is involved is it ok to update the nr state like this? In other words, is there any other NR state for which a payment would come into picture?